lib/src/facts/posix/networking_resolver.cc

```diff
@@ -67,15 +67,19 @@ namespace facter { namespace facts { namespace posix {
         if (gethostname(name.data(), name.size()) != 0) {
             LOG_WARNING("gethostname failed: %1% (%2%): hostname is unavailable.", strerror(errno), errno);
         } else {
-            // Use everything up to the first period
-            result.hostname = name.data();
-            auto pos = result.hostname.find('.');
-            if (pos != string::npos) {
-                result.hostname = result.hostname.substr(0, pos);
+            // Check for fully-qualified hostname
```

Comment:
Maps to approximately https://github.com/puppetlabs/facter/blob/2.x/lib/facter/domain.rb#L45.